packages/data/src/utils/emitter.ts

```diff
@@ -1,12 +1,20 @@
+export interface DataEmitter {
```

Comment:
One question here, do we have an established pattern whether types should be lifted to `types.ts` file or colocated with the usage? This is an example where the decision seems random to me. As weel, the refactoring could look like:

```js
import { type DataEmitter } from '../types';
```